backend/tests/controllers/collections.test.ts

```diff
@@ -0,0 +1,226 @@
+import { describe, expect, test, jest, afterEach } from "@jest/globals"
+const express = require("express")
+const bodyParser = require("body-parser");
+const app = express()
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+const CollectionsRouter = require("../../routes/collection")
+const request = require("supertest")
+
+const Collection = require("../../models/collection")
+jest.mock("../../models/collection", () => ({
+	findOne: jest.fn(),
+    create: jest.fn()
+}))
+
+const jwt = require("jsonwebtoken")
+jest.mock("jsonwebtoken", () => ({
+	verify: jest.fn()
+}))
+
+describe('getCollection tests', () =>{
+    test("Response has status 503 (can't access the database to find the collection)", async () => {
+        Collection.findOne.mockImplementationOnce(() => {
+        	return {
+            	exec: jest.fn().mockImplementationOnce(() => {return Promise.reject()})
+          	}
+        })
+        const res = await request(app.use(CollectionsRouter))
+        .get('/testowa');
+
+        expect(res.status).toMatchInlineSnapshot(`503`)
+    })
+    test("Response has status 404 (collection with provided name doesn't exist)", async () => {
+        Collection.findOne.mockImplementationOnce(() => {
+        	return {
+            	exec: jest.fn().mockImplementationOnce(() => {return Promise.resolve(null)})
+          	}
+        })
+        const res = await request(app.use(CollectionsRouter))
+        .get('/nieistniejaca');
+
+        expect(res.status).toMatchInlineSnapshot(`404`)
+    })
```

Comment:
I would write it in more like given when then manner: https://www.j-labs.pl/blog-technologiczny/given-when-then-pattern-in-unit-tests/
```suggestion
    test("Response has status 404 (collection with provided name doesn't exist)", async () => {
       // given
        Collection.findOne.mockImplementationOnce(() => {
        	return {
            	exec: jest.fn().mockImplementationOnce(() => {return Promise.resolve(null)})
          	}
        })
        
        //when
        const res = await request(app.use(CollectionsRouter))
        .get('/nieistniejaca');

        //then
        expect(res.status).toEqual(404)
        exprect(res.body.message).toEqual('Some user error message')
    })
```